numpyro/infer/util.py

```diff
@@ -365,10 +365,10 @@ def initialize_model(rng_key, model,
                                                     model_kwargs=model_kwargs)
 
     init_strategy = init_strategy if isinstance(init_strategy, partial) else init_strategy()
-    if init_strategy.func is init_to_value:
+    if (init_strategy.func is init_to_value) and not replay_model:
         init_values = init_strategy.keywords.get("values")
         unconstrained_values = transform_fn(inv_transforms, init_values, invert=True)
-        init_strategy = _init_to_unconstrained_value(values=unconstrained_values)
+        init_strategy = partial(_init_to_unconstrained_value, values=unconstrained_values)
```

Comment:
Sorry, now that I'm taking another look, I don't think I understand this change. Wouldn't `_init_to_unconstrained_value(values=unconstrained_values)` return a `partial` function?

Reply:
Okay, I see the issue. Can we just pass in `init_to_value(values=..)`?

Reply:
I was lazy to make init_to_unconstrained_value compatible with other init strategies, hence introducing the bug. :(

With init to uniform and init to unconstrained values, we don't have to trace the model in find_valid_initial_params to generate unconstrained values. With init to value, we need to trace the model to get the transforms, then using transforms to transform the values to unconstrained values.

Reply:
I see. If we called `find_valid_initial_params` with `init_to_value(values=constrained_values)`, are we going to be using the constrained values when generating the initial params?

Reply:
Yes, we still use constrained values. But we need to trace the model in that case. This extra logic only happens when we use initialize_model with init to values.